README.txt
<strong>HERMAN CHILD THEME</strong>
http://wpcanada.ca/our-themes/herman/

<strong>INSTALL</strong>
1. Upload the Herman child theme folder via FTP to your wp-content/themes/ directory. (the Genesis parent theme needs to be in the wp-content/themes/ directory as well.)
 ** OR **
2. Upload the Herman child theme folder ZIP by navigating to Appearance > Install Themes > Upload
3. Go to your WordPress dashboard and select Appearance.
4. Activate the Herman theme.
5. Inside your WordPress dashboard, go to Genesis > Theme Settings and configure them to your liking.

<strong>HOME PAGE</strong>
The Herman child themes comes with 2 home page template files:
1. <code>home.php</code>
2. <code>home_loop.php</code>

<strong>HOME PAGE USAGE</strong>
1. <code>home.php</code> == The default homepage uses the <code>home.php</code> template file. This file features 6 widget areas - Home Slider, Home Welcome, Home Left,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Home Middle, Home Right and Home Message. If no widgets are active in these widget areas the theme will display a standard blog format. The widgets used in these areas are,

Home Slider: Genesis Responsive Slider
Home Welcome, Home Message: Text widgets
Home Left, Home Middle, Home Right: Genesis Featured Page widget

2. <code>home_loop.php</code> == This template file uses the Genesis Grid Loop. If you would like to use it instead of the default <code>home.php</code> follow these steps:
a. Rename the default <code>home.php</code> file to something like <code>home_old.php</code>
b. Rename <code>home_loop.php</code> to <code>home.php</code>
The Herman child theme will now use the Genesis Grid Loop as the default layout for the home page. To revert back simply rename the 2 files. There are several items
that can be configured by editing the <code>home_loop.php</code> template file such as number of featured posts, featured image size, featured content limit etc.

<strong>CONFIGURING HOME PAGE WIDGETS</strong>
Genesis Responsive Slider: 
a. Set Maximum Slider Width to 950
b. Set Maximum Slider Height to 300
c. Select "Display post content" from the drop down menu
d. Enter "155" characters into Limit content field
e. Set Slider Excerpt Width to 25
f. For Excerpt Location vertical choose "Top" from the drop down menu
g. For Excerpt Location horizontal choose "Right" from the drop down menu
&raquo; Please see image named <code>grs-config.png</code> in the theme's images folder for visual explanation.

Genesis Featured Page
a. Select your page from the drop down menu
b. Check the box "Show Featured Image"
c. Select "home-bottom (280x150)" from the drop down menu
d. Image Alignment select "None" from the drop down menu
e. Check the box "Show Page Title"
f. Check the box "Show Page Content"
g. Enter 160 in the Content Character Limit field
h. Enter some text in the More Text field

<strong>FEATURED IMAGES</strong>
By default WordPress will create a default thumbnail image for each image you upload and the size can be specified in your dashboard under Settings > Media. In addition, the Herman
child theme creates the following thumbnail images you'll see below, which are defined (and can be modified) in the <code>functions.php file</code>. These are the recommended
thumbnail sizes that are used on the Herman child theme demo site:

grid-thumbnail (285x120)
home-bottom (280x150)
slider (950x300)
portfolio (202x140)

You may have to resize your images. If so, we recommend the very excellent Regenerate Thumbnails plugin.
<a href="http://wordpress.org/extend/plugins/regenerate-thumbnails">http://wordpress.org/extend/plugins/regenerate-thumbnails</a>

<strong>SPECIAL FEATURES</strong>
The Herman child theme comes with the following:
1. Google fonts
2. Your choice of 3 home page layouts
3. 6 different content box styles
4. Pullquotes
5. Support for Post Formats
6. Support Simple Social Icons plugin from StudioPress
7. Landing Page template
8. Portfolio Page template
9. Widgetized 404 Page

<strong>SETTING UP THE PORTFOLIO PAGE</strong>
1. Create a page called "Portfolio" or some such thing. Don't add any content to it. From the drop-down menu select the Portfolio Page template and publish the blank page.
2. Create a category called "Portfolio".
3. Create a series of posts and file them under the Portfolio category. Set a feature image for each post.
4. Add the Genesis Featured Post widget to the Portfolio widget area on the widget screen. Configure. The feature images you selected will automatically appear on the Portfolio page.

<strong>CONTENT BOXES</strong>
The custom content boxes make use of DIV classes. For example, &lt;div class="content-box-blue"&gt;some random text&lt;/div&gt;
There are 6 styles to choose from.

<strong>Pullquotes</strong>
Pullquotes are added to posts or pages by assigning a div class to the text you want to quote. Their are 2 options you can use, left-aligned and right-aligned.

For example, to use a left-aligned pullquote do this ...
&lt;div class="pullquote-left"&gt;This is a left-aligned pullquote.&lt;/div&gt;

and to use a right-aligned pullquote do this ...
&lt;div class="pullquote-right"&gt;This is a right-aligned pullquote.&lt;/div&gt;

In previous versions of this theme I added the pullquotes as a class to the blockquote but have now changed that for a couple of reasons:
1. This method avoids any potential conflicts with blockquotes.
2. The CSS is cleaner. At least I think so.

<strong>GENESIS BOX</strong>
The Genesis Box is disabled by default. To enable it, open Herman's <code>functions.php</code> file and uncomment the function. Specifically, look for this ...

<code>/** Add Genesis Box on Single Posts */
//add_action( 'genesis_after_post_content', 'include_genesis_box', 11 );
//function include_genesis_box() {
    //if ( is_single() )
    //require( CHILD_DIR.'/genesis-box.php' );
//}</code>

... and remove all of the //

<strong>PLUGINS</strong>
Genesis Responsive Slider - <a href="http://wordpress.org/extend/plugins/genesis-responsive-slider/">http://wordpress.org/extend/plugins/genesis-responsive-slider/</a>
Simple Social Icons - <a href="http://wordpress.org/extend/plugins/simple-social-icons/">http://wordpress.org/extend/plugins/simple-social-icons</a>
Regenerate Thumbnails - <a href="http://wordpress.org/extend/plugins/regenerate-thumbnails/">http://wordpress.org/extend/plugins/regenerate-thumbnails/</a>
Genesis eNews Extended - <a href="http://wordpress.org/extend/plugins/genesis-enews-extended/">http://wordpress.org/extend/plugins/genesis-enews-extended/</a>

<strong>CREDIT</strong>
1. Post Format icons courtesy StudioPress &ndash; http://www.studiopress.com/graphics/post-format-icons
2. Content Boxes by Brian Gardner &ndash; http://www.briangardner.com/genesis-content-boxes/
3. Widgetized 404 Page by David Decker &ndash; http://genesisthemes.de/en/2011-08/tutorial-widgetized-404-error-page-in-genesis/

<strong>FURTHER</strong>
1. Theme release page &ndash; http://wpcanada.ca/our-themes/herman/
2. Theme Demo &ndash; http://demo.wpcanada.ca/herman/
3. Theme download page &ndash; http://code.google.com/p/herman-theme/

<strong>BUG REPORTS</strong>
Bug reports can be filed at http://code.google.com/p/herman-theme/

Enjoy!

changelog.txt
*** Herman Changelog ***


2012.09.17 - Version 1.0

  * Initial release

2012.09.19 - version 1.0.1

  * fixed typo in functions.php relating to Google font loading

2012.09.21 - version 1.0.2
  * tweaked .sidebar padding
  * added to #inner margin

2012.10.15 - version 1.0.3
  * added separate class for pullquotes and adjusted styling

2012.12.08 - version 1.0.4
  * fixed bug in numerical page navigation

2012.12.15 - version 1.1.0
  * added styling for Page Links (paginated posts)

2013.01.06 - version 1.1.1
  * adjusted #subnav responsive styling

2013.01.13 - version 1.2.0
  * added styling for Genesis eNews Extended plugin

404.php
<?php
/**
 * 404 Not found page example
 * 
 * This helps create a custom and widgetized 404 error page template for Genesis child themes.
 *
 * @author David Decker, http://genesisthemes.de/en/
 * @link http://genesisthemes.de/en/2011-08/tutorial-widgetized-404-error-page-in-genesis/
 *
 * Note by Len, added class 'entry-content' so that styling for 404 page is inherited by theme.
 */


remove_action( 'genesis_loop', 'genesis_do_loop' );
add_action( 'genesis_loop', 'herman_404_loop_helper' );
/**
 * Add widget support for the 404 page. If no widgets active, display the default loop/message.
 *
 */
function herman_404_loop_helper() {

	if ( is_active_sidebar( '404-page' ) ) {
			echo '<div class="entry-content">';
		dynamic_sidebar( '404-page' );
			echo '</div><!-- end .entry-content -->';		
	}
	else {
		genesis_standard_loop();
	}
	
}

genesis();

home.php
<?php

add_action( 'genesis_meta', 'herman_home_genesis_meta' );
/**
 * Add widget support for homepage. If no widgets active, display the default loop.
 *
 */
function herman_home_genesis_meta() {

	if ( is_active_sidebar( 'home-slider' ) || is_active_sidebar( 'home-welcome' ) || is_active_sidebar( 'home-left' ) || is_active_sidebar( 'home-middle' ) || is_active_sidebar( 'home-right' ) || is_active_sidebar( 'home-message' ) ) {

		remove_action( 'genesis_loop', 'genesis_do_loop' );
		add_action( 'genesis_after_header', 'herman_home_loop_helper_top' );
		add_action( 'genesis_after_header', 'herman_home_loop_helper' );
		add_filter( 'genesis_pre_get_option_site_layout', '__genesis_return_full_width_content' );

	}
}

/**
 * Display widget content for "home-slider" and "welcome" sections.
 *
 */

function herman_home_loop_helper_top() {

		if ( is_active_sidebar( 'home-slider' ) ) {
			echo '<div class="slider-wrap"><div class="slider-inner">';
			dynamic_sidebar( 'home-slider' );
			echo '</div><!-- end .slider-wrap --></div><!-- end .slider-inner -->';
		}

		if ( is_active_sidebar( 'home-welcome' ) ) {
			echo '<div class="welcome-wrap"><div class="welcome-inner">';
			dynamic_sidebar( 'home-welcome' );
			echo '</div><!-- end .welcome-wrap --></div><!-- end .welcome-inner -->';
		}
		
}

/**
 * Display widget content for "home-left", "home-middle", and "home-right" sections.
 *
 */

function herman_home_loop_helper() {

		echo '<div id="home-bottom"><div class="wrap">';

		if ( is_active_sidebar( 'home-left' ) ) {
			echo '<div class="homepage-left">';
			dynamic_sidebar( 'home-left' );
			echo '</div><!-- end #homepage-left -->';
		}

		if ( is_active_sidebar( 'home-middle' ) ) {
			echo '<div class="homepage-middle">';
			dynamic_sidebar( 'home-middle' );
			echo '</div><!-- end #homepage-middle -->';
		}

		if ( is_active_sidebar( 'home-right' ) ) {
			echo '<div class="homepage-right">';
			dynamic_sidebar( 'home-right' );
			echo '</div><!-- end #homepage-right -->';
		}
		
		echo '</div><!-- end .wrap --></div><!-- end #home-bottom -->';

		if ( is_active_sidebar( 'home-message' ) ) {
			echo '<div class="message-wrap"><div class="message-inner">';
			dynamic_sidebar( 'home-message' );
			echo '</div><!-- end .message-wrap --></div><!-- end .message-inner -->';
		}
		
}

genesis();

genesis-box.php
<div id="genesis-box">
    <h3>YourDomain.com runs on the Genesis Framework</h3>
    <a href="AFFILIATE LINK GOES HERE"><img class="alignright" src="<?php echo get_stylesheet_directory_uri() ?>/images/genesis-260x125.jpg" width="260" height="125" alt="Genesis Framework" title="Check out the Genesis Framework" /></a>
    <p>Genesis empowers you to quickly and easily build incredible websites with WordPress. Whether you're a novice or advanced developer, Genesis provides the secure and search-engine-optimized foundation that takes WordPress to places you never thought it could go. It's that simple - start using <a href="AFFILIATE LINK GOES HERE">Genesis</a> now!</p>
    <p>Take advantage of the 6 default layout options, comprehensive SEO settings, rock-solid security, flexible theme options, cool custom widgets, custom design hooks, and a huge selection of child themes ("skins") that make your site look the way you want it to.  With automatic theme updates and world-class support included, Genesis is the smart choice for your WordPress website or blog.</p>
</div>


page_landing.php
<?php

/*
Template Name: Landing
*/

// Add custom body class to the head
add_filter( 'body_class', 'herman_add_body_class' );
function herman_add_body_class( $classes ) {
   $classes[] = 'herman-landing';
   return $classes;
}

// Remove header, navigation, breadcrumbs, footer widgets, footer 
add_filter( 'genesis_pre_get_option_site_layout', '__genesis_return_full_width_content' );
remove_action( 'genesis_header', 'genesis_header_markup_open', 5 );
remove_action( 'genesis_header', 'genesis_do_header' );
remove_action( 'genesis_header', 'genesis_header_markup_close', 15 );
remove_action( 'genesis_before', 'genesis_do_nav' );
remove_action( 'genesis_after_header', 'genesis_do_subnav' );
remove_action( 'genesis_after_header', 'genesis_do_breadcrumbs');
remove_action( 'genesis_before_footer', 'genesis_footer_widget_areas' );
remove_action( 'genesis_footer', 'genesis_footer_markup_open', 5 );
remove_action( 'genesis_footer', 'genesis_do_footer' );
remove_action( 'genesis_footer', 'genesis_footer_markup_close', 15 );

genesis();

functions.php
<?php
/** Start the engine */
require_once( get_template_directory() . '/lib/init.php' );

/** Localization */
load_child_theme_textdomain( 'herman', get_stylesheet_directory() . '/lib/languages' );

/** Child theme (do not remove) */
define( 'CHILD_THEME_NAME', 'Herman Theme' );
define( 'CHILD_THEME_URL', 'http://wpcanada.ca/our-themes/herman/' );

/** Load Google fonts */
add_action( 'wp_enqueue_scripts', 'herman_load_google_fonts' );
function herman_load_google_fonts() {
    wp_enqueue_style( 
    	'google-fonts', 
    	'http://fonts.googleapis.com/css?family=Lato', 
    	array(), 
    	PARENT_THEME_VERSION 
    );
}

/** Add Viewport meta tag for mobile browsers */
add_action( 'genesis_meta', 'herman_add_viewport_meta_tag' );
function herman_add_viewport_meta_tag() {
    echo '<meta name="viewport" content="width=device-width, initial-scale=1.0"/>';
}

$content_width = apply_filters( 'content_width', 610, 460, 910 );

/** Add support for structural wraps */
add_theme_support( 'genesis-structural-wraps', array(
	'header',
	'nav',
	'subnav',
	'inner',
	'footer-widgets',
	'footer'
) );

/** Add new image sizes */
add_image_size( 'grid-thumbnail', 285, 120, TRUE );
add_image_size( 'home-bottom', 280, 150, TRUE );
add_image_size( 'slider', 950, 300, TRUE );
add_image_size( 'portfolio', 202, 140, TRUE );

/** Add support for custom header */
add_theme_support( 'genesis-custom-header', array(
	'width' => 960,
	'height' => 150
) );

/** Add support for custom background */
add_theme_support( 'custom-background' );

/** Reposition the primary navigation */
remove_action( 'genesis_after_header', 'genesis_do_nav' );
add_action( 'genesis_before', 'genesis_do_nav' );

/** Reposition the breadcrumbs */
remove_action( 'genesis_before_loop', 'genesis_do_breadcrumbs' );
add_action( 'genesis_after_header', 'genesis_do_breadcrumbs' );

/** Add wrapper to breadcrumbs */
add_filter('genesis_breadcrumb_args', 'herman_breadcrumb_args');
function herman_breadcrumb_args( $args ) {
	$args['prefix'] = '<div id="breadcrumbwrapper"><div class="breadcrumb">';
	$args['suffix'] = '</div></div>';
	$args['sep'] = ' &raquo; ';
    return $args;
}

/** Add support for 3-column footer widgets */
add_theme_support( 'genesis-footer-widgets', 3 );

/** Set Genesis Responsive Slider defaults */
add_filter( 'genesis_responsive_slider_settings_defaults', 'herman_responsive_slider_defaults' );
function herman_responsive_slider_defaults( $defaults ) {
	$defaults['slideshow_height'] = '300';
	$defaults['slideshow_width'] = '960';
	return $defaults;
}

/** Add custom body class to the head */
add_filter( 'body_class', 'add_body_class' );
function add_body_class( $classes ) {
   $classes[] = 'herman';
   return $classes;
}

/** Add Genesis Box on Single Posts */
//add_action( 'genesis_after_post_content', 'include_genesis_box', 11 );
//function include_genesis_box() {
    //if ( is_single() )
    //require( CHILD_DIR.'/genesis-box.php' );
//}

/** Add post nav to single post pages */
add_action('genesis_before_comments', 'custom_post_nav');
function custom_post_nav(){
 if (is_single()) {
    echo '<div class="post-nav">';
    echo '<div class="next-post-nav">';
    echo '<span class="next">';
    echo 'Next Article';
    echo '</span>';
    echo next_post_link('%link', '%title');
    echo '</div>';
    echo '<div class="prev-post-nav">';
    echo '<span class="prev">';
    echo 'Previous Article';
    echo '</span>';
    echo previous_post_link('%link', '%title');
    echo '</div>';
    echo '</div>';
  }}

/** Add support for post formats **/
add_theme_support( 'post-formats', array( 'aside', 'audio', 'chat', 'gallery', 'image', 'link', 'quote', 'status', 'video' ) );
add_theme_support( 'genesis-post-format-images' );

/** Remove elements for post formats */
add_action( 'genesis_before_post', 'herman_remove_elements' );
function herman_remove_elements() {
	
	if ( ! current_theme_supports( 'post-formats' ) )
		return;

	// Remove if post has format
	if ( get_post_format() ) {
		remove_action( 'genesis_post_title', 'genesis_do_post_title' );
		remove_action( 'genesis_before_post_content', 'genesis_post_info' );
		remove_action( 'genesis_after_post_content', 'genesis_post_meta' );
	}

	// Add back, as post has no format
	else {
		add_action( 'genesis_post_title', 'genesis_do_post_title' );
		add_action( 'genesis_before_post_content', 'genesis_post_info' );
		add_action( 'genesis_after_post_content', 'genesis_post_meta' );
	}

}

/** Register widget areas */
genesis_register_sidebar( array(
	'id'			=> 'home-slider',
	'name'			=> __( 'Home Slider', 'herman' ),
	'description'	=> __( 'This is the homepage slider section.', 'herman' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-welcome',
	'name'			=> __( 'Home Welcome', 'herman' ),
	'description'	=> __( 'This is the homepage welcome section.', 'herman' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-left',
	'name'			=> __( 'Home Left', 'herman' ),
	'description'	=> __( 'This is the homepage left section.', 'herman' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-middle',
	'name'			=> __( 'Home Middle', 'herman' ),
	'description'	=> __( 'This is the homepage middle section.', 'herman' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-right',
	'name'			=> __( 'Home Right', 'herman' ),
	'description'	=> __( 'This is the homepage right section.', 'herman' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-message',
	'name'			=> __( 'Home Message', 'herman' ),
	'description'	=> __( 'This is the homepage message section.', 'herman' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'portfolio',
	'name'			=> __( 'Portfolio Page', 'herman' ),
	'description'	=> __( 'This is the portfolio page template.', 'herman' ),
) );
genesis_register_sidebar( array(
	'id'		=> '404-page',
	'name'		=> __( '404 Page', 'herman' ),
	'description'	=> __( 'This is the widget area of the 404 Not Found Page Template.', 'herman' ),
) );